emcc.py

```diff
@@ -1258,6 +1259,7 @@ def get_bitcode_file(input_file):
         if options.debug_level == 4 or not (final_suffix in JS_CONTAINING_SUFFIXES and options.js_opts):
           newargs.append('-g') # preserve LLVM debug info
           options.debug_level = 4
```

Comment:
@kripken this current code has the effect that running emcc with just -g will cause the debug level to be bumped up to 4 here (so we always get source maps with -g or -g4). Is that the intent? It's unclear to me why we modify the debug level here.

Reply:
I think the reason is the comment 3 lines above: when we compile from source to executable (and not to bitcode or from bitcode) and the user asks for `-g` then we give source maps. This is maybe trying too hard to be clever, but the idea is to make debug info on a single-file compilation "just work", whereas when building to bitcode in the middle, users anyhow tend to specify special link-time compilation flags, so they are more aware of the need for -g4 etc.

This might behave differently in wasm than asm.js, because in asm.js when optimizing we turn on the js optimizer - which means we lose debug info anyhow. In wasm we don't use it, so this code might go into effect when it's really silly for it. (I.e., in asm.js the js optimizer being turned on prevented this code from setting the debug level to 4.)

Overall, maybe we should just replace this code with better and simpler logic. I wouldn't worry too hard about preserving the old behavior (but we should check docs and tests).

Reply:
(But all this can be left for later of course.)

Reply:
OK, this current code and preserves the current behavior and does something reasonable for wasm, so it can probably stay for now. It probably makes sense to rethink the behavior for a post-asmjs-optimizer world.